enterprise/backend/src/metabase_enterprise/task/truncate_audit_log.clj

```diff
@@ -0,0 +1,16 @@
+(ns metabase-enterprise.task.truncate-audit-log
+  "EE implementation of the `audit-max-retention-days` setting, used to determine the retention policy for audit tables."
+  (:require
+   [metabase.models.setting :as setting]
+   [metabase.models.setting.multi-setting
+    :refer [define-multi-setting-impl]]
+   [metabase.task.truncate-audit-log :as task.truncate-audit-log]))
+
+(define-multi-setting-impl task.truncate-audit-log/audit-max-retention-days :ee
+  :getter (fn []
+            (let [env-var-value (setting/get-value-of-type :integer :audit-max-retention-days)]
+              (cond
+                (nil? env-var-value)  ##Inf
+                (zero? env-var-value) ##Inf
+                (< env-var-value 30)  30
```

Comment:
As a user, I can set this value to a value in `[1,29]` via env var, right? I could have missed it, but I don't see anywhere where we give feedback that "Hey! you entered 25, but we are going to treat it as 30, since that is the right way to handle it.". It would be kind of confusing if I set some value but "metabase is ignoring it".

Should the app fail to start in that case? Or what do you think of printing a warning for it?

Reply:
Yeah fair point. I'm hesitant to make this a startup-blocking thing but definitely a warning in the logs when the setting is fetched is warranted, I'll add this